content/automate/ManageIQ/Cloud/VM/Provisioning/StateMachines/VMProvision_VM.class/__methods__/retry_placement.rb

```diff
@@ -9,12 +9,25 @@
   $evm.log(:error, "miq_provision object not provided")
   exit(MIQ_STOP)
 end
-status = $evm.inputs['status']
-$evm.root['ae_result'] = 'restart' 
-$evm.root['ae_next_state'] = 'Placement'
-$evm.log("info", "Provisioning #{prov.get_option(:vm_target_name)} failed, retrying placement.")
+
+DEFAULT_RETRIES = 3
+max_placement_retries = $evm.inputs['retries'] || DEFAULT_RETRIES
+if prov.message.include?("An error occurred while provisioning Instance")
```

Comment:
@mansam 
Is this a good enough message to capture all cloud placement errors.
Do you want to expose the "placement error message" as a passed in value from $evm.inputs so someone can customize it if they want
```
DEFAULT_PLACEMENT_ERROR_MESSAGE = "An error occurred while provisioning Instance"
placement_error_message = $evm.inputs['placement_error_message'] || DEFAULT_PLACEMENT_ERROR_MESSAGE
```